server.go

```diff
@@ -202,20 +188,63 @@ func getRequestBody(input string) (*http.Request, error) {
 	inputs = append(inputs, data)
 	reqBody := RequestBody{inputs}
 	fmt.Print(reqBody)
-	//reqBodyBytes, err := json.Marshal(reqBody)
-	//if err != nil {
-		//return nil, err
-	//}
 
 	b := new(bytes.Buffer)
-  err := json.NewEncoder(b).Encode(reqBody)
+	err := json.NewEncoder(b).Encode(reqBody)
 	req, err := http.NewRequest("POST", clarifaiApi, b)
 	fmt.Print(req)
 	if err != nil {
 		return nil, err
 	}
 
-	req.Header.Add("Authorization", `Bearer ThzzqYyVARtJdLbTQDwfRpa1FOk8w6`)
+	req.Header.Add("Authorization", `Bearer ` + clarifai_token.Token)
 	req.Header.Add("Content-Type", `application/json`)
 	return req, nil
 }
+
+func getAccesToken() AccessToken {
+	CLARIFAI_ID := os.Getenv("CLARIFAI_CLIENT_ID")
+	CLARIFAI_SECRET_KEY := os.Getenv("CLARIFAI_SECRET_KEY")
+
+	responseData := authRequest(CLARIFAI_ID, CLARIFAI_SECRET_KEY)
+	var token AccessToken
+	rawIn := json.RawMessage(responseData)
+
+	bytes, err := rawIn.MarshalJSON()
+	if err != nil {
+		fmt.Print("\nERROR Parsing authentication response")
+		panic(err)
+	}
+
+	err = json.Unmarshal(bytes, &token)
+	if err != nil {
+		fmt.Print("\nERROR Parsing authentication json string")
+		panic(err)
+	}
+	fmt.Print("\nRESULTS: ")
+	fmt.Print(token.Token)
+	fmt.Print(token.ExpiresIn)
+
+	return token
+}
+
+func authRequest(CLIENT_ID string, SECRET_KEY string) []byte {
+
+	resp, err := http.PostForm(clarifyAuth, url.Values{"grant_type": {"client_credentials"},
+		"client_id": { CLIENT_ID },
+		"client_secret": { SECRET_KEY } })
+
+	fmt.Print("\nAuthentication Request constructed and sent")
+
+	if err != nil {
+		panic(err)
```

Comment:
^ You are right, but this is not an error caused by the user. If this happens there's nothing the user can do, as our app is literally not usable (we can't do anything without an access token to clarifai). We can throw a 500 internal error msg but.. we probably shouldn't even reach this situation in the first place.

@DariaSova I have a couple questions:

1 - Why do we generate an access token per request? I thought this token should be global to our server, and every request will use this same token until it expires

2 - How's the expiration works with this token? Does it expire after some number of requests requests? After some time? We should probably have some kind of an alerting mechanism that will notify us when the token is about to expire, so we can avoid this situation.